app/models/pxe_server.rb

```diff
@@ -44,7 +43,8 @@ def default_pxe_image_for_windows
   def synchronize_advertised_images
     pxe_menus.each(&:synchronize)
     sync_windows_images
-    clear_association_cache
+    pxe_menus.reload
+    windows_images.reload
```

Comment:
need to reload all of the associations here - which is why we did a full reload

Reply:
Ok, I'll see if this breaks the specs where we call this on an object with changed values not yet persisted.  That's why I removed the object reload in favor of targetted association reloads.

Reply:
It should be relatively trivial (famous last words) to set up a spec that does a before -> sync -> after and verfies the methods return the right things

Reply:
@Fryguy Were you thinking of something like walking all reflections?

Seems like overkill.  I don't see existing tests for `synchronize_advertised_images`.  I really don't trust `object.reload` to wipe the associations since it also discards changes and blows up of `object` isn't yet persisted.

I can do this but I really don't like it:

```diff
diff --git a/app/models/pxe_server.rb b/app/models/pxe_server.rb
index a1ab1185b5..9e2c86a7ef 100644
--- a/app/models/pxe_server.rb
+++ b/app/models/pxe_server.rb
@@ -43,8 +43,10 @@ class PxeServer < ApplicationRecord
   def synchronize_advertised_images
     pxe_menus.each(&:synchronize)
     sync_windows_images
-    pxe_menus.reload
-    windows_images.reload
+    self.class.reflections.keys.each do |r|
+      reloading_method = "reload_#{r}"
+      respond_to?(reloading_method) ? send(reloading_method) : send(r).reload
+    end
     update_attribute(:last_refresh_on, Time.now.utc)
   end
```

A hardcoded list is probably more explicit.  I'm not sure what would change under us but I don't really follow the pxe_menu#synchronize logic for changing class types, etc.


```diff
diff --git a/app/models/pxe_server.rb b/app/models/pxe_server.rb
index a1ab1185b5..a41c2c8df3 100644
--- a/app/models/pxe_server.rb
+++ b/app/models/pxe_server.rb
@@ -43,8 +43,16 @@ class PxeServer < ApplicationRecord
   def synchronize_advertised_images
     pxe_menus.each(&:synchronize)
     sync_windows_images
+
+    authentications.reload
+    taggings.reload
+    tags.reload
     pxe_menus.reload
+    pxe_images.reload
+    advertised_pxe_images.reload
+    discovered_pxe_images.reload
     windows_images.reload
+
     update_attribute(:last_refresh_on, Time.now.utc)
   end
```

Reply:
No, I was thinking straight reload, and then a simple spec that proves reload works by verifying each of the associations in the spec.